File: 412_Fizz_Buzz.py
_author_ = 'jake'
_project_ = 'leetcode'

# https://leetcode.com/problems/fizz-buzz/
# Write a program that outputs the string representation of numbers from 1 to n.
# But for multiples of three it should output “Fizz” instead of the number and for the multiples of five output “Buzz”.
# For numbers which are multiples of both three and five output “FizzBuzz”.

# Test if divisible by 15 or 5 or 3 or none of these.
# Time - O(n)
# Space - O(n)

class Solution(object):
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        result = []

        for i in range(1, n + 1):
            if i % 15 == 0:
                result.append("FizzBuzz")
            elif i % 3 == 0:
                result.append("Fizz")
            elif i % 5 == 0:
                result.append("Buzz")
            else:
                result.append(str(i))

        return result

File: java/412_Fizz_Buzz.java
/*
https://leetcode.com/problems/fizz-buzz/
Write a program that outputs the string representation of numbers from 1 to n.
But for multiples of three it should output “Fizz” instead of the number and for the multiples of five output “Buzz”.
For numbers which are multiples of both three and five output “FizzBuzz”.

Time - O(n)
Space - O(n)
*/

public class Solution {
    public List<String> fizzBuzz(int n) {

        ArrayList results = new ArrayList();

        for (int i = 1; i <= n; ++i) {
            if (i % 15 == 0)
                results.add("FizzBuzz");
            else if (i % 5 == 0)
                results.add("Buzz");
            else if (i % 3 == 0)
                results.add("Fizz");
            else
                results.add(String.valueOf(i));
        }
        return results;
    }
}
